RTech.CoffeeMachine.Api/BrewCoffee/Filters/BrewStatusUnavailableFilter.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Caching.Memory;
+using RTech.CoffeeMachine.Api.BrewCoffee.Controllers;
+
+namespace RTech.CoffeeMachine.Api.BrewCoffee.Filters;
+
+public class BrewStatusUnavailableFilter : ActionFilterAttribute
+{
+    // Interviewer note: MemoryCache from Microsoft.Extensions.Caching.Abstractions is thread safe
+    private readonly IMemoryCache _cache;
+    public static readonly string CACHE_KEY_NUM_REQUESTS = $"{nameof(CoffeeMachineController.GetBrewStatus)}:numRequests";
+
+    public BrewStatusUnavailableFilter(IMemoryCache cache)
+    {
+        _cache = cache;
+        if (!_cache.TryGetValue<long>(CACHE_KEY_NUM_REQUESTS, out var _))
+        {
+            _cache.Set<long>(CACHE_KEY_NUM_REQUESTS, 0);
+        }
+    }
+
+    public override void OnActionExecuting(ActionExecutingContext context)
+    {
+        // Interviewer note: I'm assuming that on april 1st, every 5th request still returns 418
```

Comment:
would be a shame if only 80% of people to get an april fools joke..

jokes aside, this is an assumption i've made (ideally I would clarify this ambiguous requirement with the product owner/stakeholder, since it saves developer effort)